### PR TITLE
Refactor redirect routes

### PR DESCRIPTION
**Pull request by isoos**
- moved domain selection to `handlers.dart`
- moved most redirect route config to `redirects.dart`

### CODE REVIEW

File: app/lib/frontend/handlers.dart [hunk 1]
```diff
@@ -29,21 +30,36 @@ void _logPubHeaders(shelf.Request request) {
...
   final pubDartlangOrgHandler = PubDartlangOrgService().router.handler;
   final pubSiteHandler = PubSiteService(shelfPubApi).router.handler;
   return (shelf.Request request) async {
     _logPubHeaders(request);
 
+    // legacy dartdocs.org URLs
+    final host = request.requestedUri.host;
+    if (host == 'www.dartdocs.org' || host == 'dartdocs.org') {
+      final rs = await legacyDartdocHandler(request);
+      return rs ?? shelf.Response.notFound('Not Found.');
```
Comment (jonasfj) [inline app/lib/frontend/handlers.dart]: Let's move `shelf.Response.notFound('Not Found.')` into `LegacyDartdocService`.

Using a catch-all rule at the end of the class..

File: app/lib/frontend/handlers.dart [hunk 2]
```diff
@@ -53,3 +69,11 @@ shelf.Handler createAppHandler(shelf.Handler shelfPubApi) {
     return formattedNotFoundHandler(request);
   };
 }
+
+bool _shouldRedirectToPubDev(String path) {
+  if (path.startsWith('/api/')) return false;
+  if (path.endsWith('.tar.gz')) return false;
+  if (path.endsWith('.json')) return false;
+  if (path == '/debug') return false;
```
Comment (jonasfj) [inline app/lib/frontend/handlers.dart]: Why `/debug` ?

File: app/lib/frontend/handlers.dart [hunk 3]
```diff
@@ -53,3 +69,11 @@ shelf.Handler createAppHandler(shelf.Handler shelfPubApi) {
     return formattedNotFoundHandler(request);
   };
 }
+
+bool _shouldRedirectToPubDev(String path) {
+  if (path.startsWith('/api/')) return false;
+  if (path.endsWith('.tar.gz')) return false;
+  if (path.endsWith('.json')) return false;
```
Comment (jonasfj) [inline app/lib/frontend/handlers.dart]: Why don't we redirect these? Can we maybe leave a comment at the end of each of these lines?

File: app/lib/frontend/handlers.dart (hunk 2 above)
Comment (isoos) [reply app/lib/frontend/handlers.dart]: good catch, no reason

**Pull request merged**